src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -119,6 +120,14 @@
     boolean isChatColorsEnabled();
 
     /**
+     * Simulates a chat message from a player.
+     *
+     * @param message The message to send
+     * @return The event thrown from the message
+     */
+    MessageChannelEvent.Chat chat(Text message);
```

Comment:
That was one part I wish to change, I'd need to discuss it with people first - I wanted to move chat handling over to common, and handle it properly there as a Text.

As well as that, I want to discuss how it'd be handled on a client plugin - which is why I've currently put the code in EntityPlayerMP.